tensorflow/compiler/tf2tensorrt/common/datavec.h

```diff
@@ -19,14 +19,23 @@ limitations under the License.
 #include <vector>
 
 #include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/platform/types.h"
 
 namespace tensorflow {
 namespace tensorrt {
 
 // Input/output data format for OpConverterTest::BuildAndRun().
 struct InputOutputData {
-  void* Buffer() const {
-    return const_cast<char*>(tensor.tensor_data().data());
+  int8* Buffer() {
+    int64 num_elem =
+        (tensor.NumElements() * DataTypeSize(tensor.dtype())) / sizeof(int8);
+    return tensor.bit_casted_shaped<int8, 1>({num_elem}).data();
+  }
+
+  const int8* Buffer() const {
+    int64 num_elem =
+        (tensor.NumElements() * DataTypeSize(tensor.dtype())) / sizeof(int8);
+    return tensor.bit_casted_shaped<int8, 1>({num_elem}).data();
```

Comment:
From my point of view, we are a consumer of this `tensorflow::Tensor` data structure. It does not explicitly expose a non-const ` pointer access method to the underlying buffer. I would tend to think that there is a reason for this, given how prevalent `tensorflow::Tensor` is. I looked in the `tensor.h` header file and didn't see any comments saying something like this would be OK. I would err on the side of const-correctness. Do you disagree? Maybe this is a common shortcut? Why don't they just put that in the ITensor* class then?

Reply:
the member function is not currently used. Are you planning to use it?
I don't understand "I would err on the side of const-correctness.", can you explain?

Reply:
I did not realize this was dead code. I think this was a rebase mistake. What I meant to correct is that there is code in `trt_engine_utils.cc` which is `const_cast`'ing the `tensor` member of this struct, so adding these methods could be used there. 

In any case, if it's dead code, then I'm going to delete it and will cleanup `trt_engine_utils.cc` in another PR. 

> I don't understand "I would err on the side of const-correctness.", can you explain?
I meant that I would be very cautious about violating const qualifier rules. The reason const-cast is used here is because the author wanted to get a `void*` rather than a `const void*` from the `Buffer()` method. They obviously couldn't figure out how to get a non-const `void*` from the `ITensor` class (which is what I replaced the function with). The API is non-obvious. So they just `const_cast`'d the `const void*` that they could get with `tensor.data()`, which is a very bad idea. There's no good reason to use `const_cast` when equivalent code that obeys const qualifier rules exists, which is why I corrected the `const_cast` in the ITensorProxy class case a few PRs back. For all we know, casting away `const` qualifier violates some synchronization mechanism or performance optimization in ITensor. Using const_cast to make code smaller seems like a recipie for disaster